src/Compare.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Semver;
+
+use Composer\Semver\Constraint\VersionConstraint;
+use InvalidArgumentException;
+
+class Compare
+{
+    /**
+     * Whether $version1 > $version2.
+     *
+     * @param string $version1
+     * @param string $version2
+     *
+     * @return bool
+     */
+    public static function greaterThan($version1, $version2)
+    {
+        return self::compare($version1, '>', $version2);
+    }
+
+    /**
+     * Whether $version1 >= $version2.
+     *
+     * @param string $version1
+     * @param string $version2
+     *
+     * @return bool
+     */
+    public static function greaterThanEqualTo($version1, $version2)
+    {
+        return self::compare($version1, '>=', $version2);
+    }
+
+    /**
+     * Whether $version1 < $version2.
+     *
+     * @param string $version1
+     * @param string $version2
+     *
+     * @return bool
+     */
+    public static function lessThan($version1, $version2)
+    {
+        return self::compare($version1, '<', $version2);
+    }
+
+    /**
+     * Whether $version1 <= $version2.
+     *
+     * @param string $version1
+     * @param string $version2
+     *
+     * @return bool
+     */
+    public static function lessThanEqualTo($version1, $version2)
+    {
+        return self::compare($version1, '<=', $version2);
+    }
+
+    /**
+     * Whether $version1 == $version2.
+     *
+     * @param string $version1
+     * @param string $version2
+     *
+     * @return bool
+     */
+    public static function equal($version1, $version2)
+    {
+        return self::compare($version1, '==', $version2);
+    }
+
+    /**
+     * Whether $version1 != $version2.
+     *
+     * @param string $version1
+     * @param string $version2
+     *
+     * @return bool
+     */
+    public static function notEqual($version1, $version2)
+    {
+        return self::compare($version1, '!=', $version2);
+    }
+
+    /**
+     * Evaluate $version1 $comparison $version2.
+     *
+     * @param string $version1
+     * @param string $comparison Operator like ">" or "<=", etc.
+     * @param string $version2
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return bool
+     */
+    public static function compare($version1, $comparison, $version2)
+    {
+        if (!in_array($comparison, VersionConstraint::getSupportedOperators())) {
+            throw new InvalidArgumentException("'$comparison' is not a supported operator");
```

Comment:
We usually don't import classes that exist in the global namespace but use them directly instead (so prefix with a `\`). Also try to avoid `"`, better to go with something like:

```
sprintf(
  'Operator "%s" not supported, expected one of: %s',
  $comparison,
  implode(', ', VersionConstraint::getSupportedOperators())
);
```